Userland/Libraries/LibVT/TerminalWidget.cpp

```diff
@@ -1155,4 +1153,34 @@ void TerminalWidget::set_font_and_resize_to_fit(const Gfx::Font& font)
     set_font(font);
     resize(widget_size_for_font(font));
 }
+
+// Used for sending data that was not directly typed by the user.
+// This basically wraps the code that handles sending the escape sequence in bracketed paste mode.
+void TerminalWidget::send_non_user_input(const ReadonlyBytes& bytes)
+{
+    constexpr StringView leading_control_sequence = "\e[200~";
+    constexpr StringView trailing_control_sequence = "\e[201~";
+
+    int nwritten;
+    if (m_terminal.needs_bracketed_paste()) {
+        // We do not call write() separately for the control sequences and the data,
+        // because that would present a race condition where another process could inject data
+        // to prematurely terminate the escape. Could probably be solved by file locking.
+        Vector<u8> output;
+        output.ensure_capacity(leading_control_sequence.bytes().size() + bytes.size() + trailing_control_sequence.bytes().size());
+
+        // HACK: We don't have a `Vector<T>::unchecked_append(Span<T> const&)` yet :^(
+        [[maybe_unused]] auto rc1 = output.try_append(leading_control_sequence.bytes().data(), leading_control_sequence.bytes().size());
+        [[maybe_unused]] auto rc2 = output.try_append(bytes.data(), bytes.size());
+        [[maybe_unused]] auto rc3 = output.try_append(trailing_control_sequence.bytes().data(), trailing_control_sequence.bytes().size());
```

Comment:
There's no real reason to use `try_append()` here, there's an `append(const T*, size_t)` that does the same thing and asserts on failure.
your comments will still be valid.

Sidenote, can we not just write these separately?

Reply:
I'm not sure about the separate writing part. That could be a data race where some other process could maliciously inject the end marker.

I'll fix the `tri_append()` stuff.

Reply:
That's fair, I guess.